eo-maven-plugin/src/main/java/org/eolang/maven/DownloadDepsMojo.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.function.BiConsumer;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.cactoos.list.ListOf;
+
+/**
+ * Downloads dependencies.
+ *
+ * @since 0.38
+ */
+@Mojo(
+    name = "deps",
+    defaultPhase = LifecyclePhase.PROCESS_SOURCES,
+    threadSafe = true
+)
+public final class DownloadDepsMojo extends SafeMojo {
+
+    /**
+     * Dependencies to download.
+     */
+    private static final Collection<Dependency> DEPS = new ListOf<>(
+        new DepBuilder().withGroupId("net.java.dev.jna").withArtifactId("jna")
+            .withVersion("5.14.0").withScope("compile").build()
+    );
+
+    /**
+     * Directory where classes are stored in target.
+     * @checkstyle MemberNameCheck (8 lines)
+     */
+    @Parameter(
+        defaultValue = "${project.build.directory}/classes",
+        readonly = true,
+        required = true
+    )
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private File classesDir;
+
+    /**
+     * The central.
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    private BiConsumer<Dependency, Path> central;
+
+    @Override
+    void exec() throws IOException {
+        if (this.central == null) {
+            this.central = new Central(this.project, this.session, this.manager);
+        }
+        for (final Dependency dep: DownloadDepsMojo.DEPS) {
+            this.central.accept(dep, this.classesDir.toPath());
+        }
+    }
+
+    /**
+     * Object for building {@link Dependency} inplace.
+     *
+     * @since 0.38
+     */
+    private static class DepBuilder {
```

Comment:
@c71n93 it would be better not to use builder here. Instead I would suggest to use delayed `Consumer` which would build `Dependency` only when we "touch" it. And it would be better not to use word "builder" in the name of the class

Reply:
@maxonfjvipon I don't understand. Could you please provide any example of delayed `Consumer`?